addons/test/misra/misra-test.c

```diff
@@ -395,6 +395,13 @@ uint8_t misra_8_4_buf2[24]; // 8.4
 typedef struct { uint16_t a; uint16_t b; } misra_8_4_struct;
 extern misra_8_4_struct bar[42];
 misra_8_4_struct bar[42]; // compliant
+extern uint16_t misra_8_4_speed = 6000u; //8.4
+uint8_t misra_8_4_pressure = 101u; //8.4
+int32_t misra_8_4_ext_val2;
+int32_t misra_8_4_ext_val2 = 3; // compliant
+int32_t misra_8_4_ext_val3;
+misra_8_4_ext_val3 = 3; // compliant
```

Comment:
This is not valid code I believe. You can initialize a variable in global scope but not have standalone assignments like this. In the Cppcheck tokenlist it's possible for assignments though but then the original C code is always a variable initialization and then the isSplittedVarDeclEq flag is set..